test/lib/render-helpers-navigate.js

```diff
@@ -0,0 +1,193 @@
+/* eslint-disable no-unused-vars -- ESLint is confused here */
+import React, { useMemo, useState } from 'react';
+import { Provider } from 'react-redux';
+import { render } from '@testing-library/react';
+import { renderHook } from '@testing-library/react-hooks';
+import { userEvent } from '@testing-library/user-event';
+import { MemoryRouter } from 'react-router-dom';
+import { CompatRouter } from 'react-router-dom-v5-compat';
+import PropTypes from 'prop-types';
+import configureStore from '../../ui/store/store';
+import { I18nContext, LegacyI18nProvider } from '../../ui/contexts/i18n';
+import { LegacyMetaMetricsProvider } from '../../ui/contexts/metametrics';
+import { getMessage } from '../../ui/helpers/utils/i18n-helper';
+import * as en from '../../app/_locales/en/messages.json';
+import { setupInitialStore } from '../../ui';
+import Root from '../../ui/pages';
+
+export const I18nProvider = (props) => {
+  const { currentLocale, current, en: eng } = props;
+
+  const t = useMemo(() => {
+    return (key, ...args) =>
+      getMessage(currentLocale, current, key, ...args) ||
+      getMessage(currentLocale, eng, key, ...args);
+  }, [currentLocale, current, eng]);
+
+  return (
+    <I18nContext.Provider value={t}>{props.children}</I18nContext.Provider>
+  );
+};
+
+I18nProvider.propTypes = {
+  currentLocale: PropTypes.string,
+  current: PropTypes.object,
+  en: PropTypes.object,
+  children: PropTypes.node,
+};
+
+I18nProvider.defaultProps = {
+  children: undefined,
+};
+
+const createProviderWrapper = (store, pathname = '/') => {
+  const Wrapper = ({ children }) =>
+    store ? (
+      <Provider store={store}>
+        <MemoryRouter initialEntries={[pathname]}>
+          <CompatRouter>
+            <I18nProvider currentLocale="en" current={en} en={en}>
+              <LegacyI18nProvider>
+                <LegacyMetaMetricsProvider>
+                  {children}
+                </LegacyMetaMetricsProvider>
+              </LegacyI18nProvider>
+            </I18nProvider>
+          </CompatRouter>
+        </MemoryRouter>
+      </Provider>
+    ) : (
+      <MemoryRouter initialEntries={[pathname]}>
+        <CompatRouter>
+          <I18nProvider currentLocale="en" current={en} en={en}>
+            <LegacyI18nProvider>
+              <LegacyMetaMetricsProvider>{children}</LegacyMetaMetricsProvider>
+            </LegacyI18nProvider>
+          </I18nProvider>
+        </CompatRouter>
+      </MemoryRouter>
+    );
```

Comment:
nit: What do you think about extracting the repeated component here into its own variable? This would prevent code drift between the two conditional branches.

```suggestion
const createProviderWrapper = (store, pathname = '/') => {
  const Wrapper = ({ children }) => {
    const container = (
      <MemoryRouter initialEntries={[pathname]}>
        <CompatRouter>
          <I18nProvider currentLocale="en" current={en} en={en}>
            <LegacyI18nProvider>
              <LegacyMetaMetricsProvider>{children}</LegacyMetaMetricsProvider>
            </LegacyI18nProvider>
          </I18nProvider>
        </CompatRouter>
      </MemoryRouter>
    );
    return store ? (
      <Provider store={store}>{container}</Provider>
    ) : (
      { container }
    );
```